todo2.js

```diff
@@ -0,0 +1,82 @@
+var tasksArray = [];
+
+document.addEventListener("DOMContentLoaded", function () {
+  let taskInput = document.getElementById("input");
+  let taskList = document.getElementById("tasks-list");
+  let addButton = document.getElementById("add-button");
+
+  addButton.addEventListener("click", function () {
+    addTask(taskInput, taskList);
+  });
+
+  taskInput.addEventListener("keydown", function (event) {
+    if (event.key === "Enter") {
+      addTask(taskInput, taskList);
+    }
+  });
+});
+
+function addTask(taskInput, taskList) {
+  let taskObject = {
+    task: taskInput.value,
+    completed: false,
+  };
+  tasksArray.push(taskObject);
+  // create new task
+  let [checkBox, li, eraseButton] = createTask(taskObject);
+  // add task to list
+  taskList.appendChild(li).appendChild(checkBox);
```

Comment:
move this piece of logic to the create task function.
No need to return the elements and then append them.